packages/fast-components/src/dialog/dialog.ts

```diff
@@ -0,0 +1,145 @@
+import { attr, FastElement } from "@microsoft/fast-element";
+import { keyCodeEscape, keyCodeTab } from "@microsoft/fast-web-utilities";
+import tabbable from "tabbable";
+
+export class Dialog extends FastElement {
+    @attr
+    public modal: boolean = true;
+
+    @attr
+    public hidden: boolean = false;
+
+    @attr({ attribute: "trap-focus" })
```

Comment:
is focus trapping independent of the dialog's modal value?

Reply:
Right now it is. The thought was to provide a means for people to opt-out, even if that means in modal scenarios. The intent was based on this from the W3C:

>Like non-modal dialogs, modal dialogs contain their tab sequence. That is, Tab and Shift + Tab do not move focus outside the dialog. However, unlike most non-modal dialogs, modal dialogs do not provide means for moving keyboard focus outside the dialog window without closing the dialog.

Currently the coded behavior is that both contain their tab sequence (first two sentences refer to this). If we want to ensure that never happens with a modal, I'm good with that, but this was intentional and meant to somewhat spark the conversation on "how flexible" we want to be.